base/dict.jl

```diff
@@ -344,6 +344,7 @@ ht_keyindex2!(h::Dict, key) = ht_keyindex2_shorthash!(h, key)[1]
     h.vals[index] = v
     h.count += 1
     h.age += 1
+    isslotmissing(h, index) && h.ndel -= 1
```

Comment:
I don't expect this will predict well in many cases, so perhaps better make it branch free? Also bounds access would have failed earlier if `index` is not in bounds, so regardless of caller `@inbounds` we can apply it here (incidentally will inbounds propagation recurse to next level in the call chain?).
```suggestion
    h.ndel = ifelse(@inbounds(isslotmissing(h, index)), h.ndel - 1, h.ndel)
```